### DISCUSSION TITLE
Validation failedfor generic form field

### DISCUSSION THREAD

**Discussion by vvusts** [react-hook-form/react-hook-form#7302] [Q&A] [answered]
I am trying to make generic component for text input for react hook form.
Here is the link to sandbox:
https://codesandbox.io/s/eager-https-0osjz?file=/src/App.tsx
Problem is with validation and it started when I moved code to separate component.
What I did wrong here?
I suspect that I make some issue with TS as I am using it first time.
Also I am nut sure do I need `<FormProvider {...methods}>` at all?

**Comment by jorisre** [comment, marked as answer]
Answered on stackoverflow: https://stackoverflow.com/questions/70370616/validation-with-yup-failed-for-generic-react-hook-form-component/70375798?noredirect=1#comment124402672_70375798
